core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -221,7 +221,8 @@ public BytesToBytesMap(
       SparkEnv.get() != null ? SparkEnv.get().blockManager() :  null,
       SparkEnv.get() != null ? SparkEnv.get().serializerManager() :  null,
       initialCapacity,
-      0.70,
+      // In order to re-use the longArray for sorting, the load factor cannot be larger than 0.5.
+      0.5,
```

Comment:
given that this may cause `BytesToBytesMap` to spill to disk more, do we have a sense of the performance implications of this change? Also, can we make this configurable (so that if `spark.sql.sort.enableRadixSort` = false, users can explicitly set the loadFactor to 0.7)?